File: src/crtp/crtp/packers/console_packer.py
import struct 

from .packer import Packer

class ConsolePacker(Packer):
    PORT_CONSOLE = 0

    CHANNEL_CONSOLE = 0

    def __init__(self, CrtpPacker):
        super().__init__(CrtpPacker, self.PORT_CONSOLE)

    def consolepacket(self):
        data = struct.pack('<')
        return self._prepare_packet(channel=self.CHANNEL_CONSOLE,
                                    data=data)

File: src/crtp_cpp/crtp_packer/console_packer.cpp
#include "crtp_packer/console_packer.hpp"

#define PORT_CONSOLE 0
#define CHANNEL_CONSOLE 0

ConsolePacker::ConsolePacker()
    : CrtpPacker(PORT_CONSOLE) {}

CrtpPacket ConsolePacker::prepare_packet(const std::vector<uint8_t>& data) {
    return CrtpPacker::prepare_packet(CHANNEL_CONSOLE, data);
}

CrtpPacket ConsolePacker::consolepacket() {
    std::vector<uint8_t> data; // Empty vector for struct.pack("<")
    return prepare_packet(data);
}